pkg/kubelet/config/config.go

```diff
@@ -323,30 +321,20 @@ func (s *podStorage) seenSources(sources ...string) bool {
 func filterInvalidPods(pods []*v1.Pod, source string, recorder record.EventRecorder) (filtered []*v1.Pod) {
 	names := sets.String{}
 	for i, pod := range pods {
-		var errlist field.ErrorList
-		// TODO: remove the conversion when validation is performed on versioned objects.
-		internalPod := &api.Pod{}
-		if err := k8s_api_v1.Convert_v1_Pod_To_api_Pod(pod, internalPod, nil); err != nil {
-			glog.Warningf("Pod[%d] (%s) from %s failed to convert to v1, ignoring: %v", i+1, format.Pod(pod), source, err)
-			recorder.Eventf(pod, v1.EventTypeWarning, "FailedConversion", "Error converting pod %s from %s, ignoring: %v", format.Pod(pod), source, err)
-			continue
-		}
-		if errs := validation.ValidatePod(internalPod); len(errs) != 0 {
-			errlist = append(errlist, errs...)
-			// If validation fails, don't trust it any further -
-			// even Name could be bad.
+		var errList field.ErrorList
+		// no need to convert to api.Pod for validation again
+		// since we only care about pod name here
+		name := kubecontainer.GetPodFullName(pod)
+		if names.Has(name) {
+			// TODO: when validation becomes versioned, this gets a bit
+			// more complicated.
+			errList = append(errList, field.Duplicate(field.NewPath("metadata", "name"), pod.Name))
 		} else {
-			name := kubecontainer.GetPodFullName(pod)
-			if names.Has(name) {
-				// TODO: when validation becomes versioned, this gets a bit
-				// more complicated.
-				errlist = append(errlist, field.Duplicate(field.NewPath("metadata", "name"), pod.Name))
-			} else {
-				names.Insert(name)
-			}
+			names.Insert(name)
 		}
-		if len(errlist) > 0 {
-			err := errlist.ToAggregate()
+
+		if len(errList) > 0 {
+			err := errList.ToAggregate()
 			glog.Warningf("Pod[%d] (%s) from %s failed validation, ignoring: %v", i+1, format.Pod(pod), source, err)
```

Comment:
The warning could be more specific since there is not much validation being preformed here.